specs/src/specs/data_square_layout.md

```diff
@@ -25,28 +25,41 @@ First, we impose some ground rules:
 
 To that end, we impose some additional rules onto _blobs only_: blobs must be placed is a way such that both the transaction sender and the block producer can be held accountable—a necessary property for e.g. fee burning. Accountable in this context means that
 
 1. The transaction sender must pay sufficient fees for blob inclusion.
 1. The block proposer cannot claim that a blob was included when it was not (which implies that a transaction and the blob it pays for must be included in the same block). In addition all blobs must be accompanied by a PayForBlob transaction.
 
-Specifically, blobs must begin at a new share. We note a nice property from this rule: if the transaction sender knows 1) `k`, the size of the matrix, 2) the starting location of their blob in a row, and 3) the length of the blob (they know this since they are sending the blob), then they can actually compute a sequence of roots to _subtrees in the row NMTs_. More importantly, anyone can compute this, and can compute _the simple Merkle root of these subtree roots_.
+Specifically, a `MsgPayForBlobs` must include a `ShareCommitment` over the contents of each blob it is paying for. If the transaction sender knows 1) `k`, the size of the matrix, 2) the starting location of their blob in a row, and 3) the length of the blob (they know this since they are sending the blob), then they can actually compute a sequence of roots to _subtrees in the row NMTs_. Taking _the simple Merkle root of these subtree roots_ provides us with the `ShareCommitment` that gets included in `MsgPayForBlobs`. Using subtree roots instead of all the leafs makes blob inclusion proofs smaller.
 
-This, however, requires the block producer to interact with the transaction sender to provide them the starting location of their blob, so that the sender can sign over the commitment based on that starting location. This can be done selectively, but is not ideal as a default for e.g. end-user wallets.
+![subtree roots](./figures/subtree_roots.svg)
 
-### Blob Share Commitment Rules
+Understanding 1) and 2) would usually require interaction with the block proposer. To make the possible starting locations of blobs sufficiently predictable and to make `ShareCommitment` independent of `k`, we impose an additional rule. The blob must start at an index that is either:
 
-To make the possible starting locations of blobs sufficiently predictable and constrained such that users can deterministically compute subtree roots, needed for the `ShareCommitment`s within a PFB, without the need to interact with the block proposer, we impose one additional rule:
+1. A multiple of the `SubtreeWidth`.
+2. A multiple of `k`, the size of the square.
```

Comment:
[blocking] I don't think the blob must start at a multiple of `k` 

```suggestion
```

Reply:
It did. Whenever padding was large enough that it would overflow into the next row, instead of starting there we would start at the beginning of the row.

I cleaned it up but in a way this rule still holds. For the implementation, instead of making it a multiple of `k`, we say that the blob needs to be able to fit in a square and since the `SubtreeWidth` is the `BlobMinSquareSize` when the `SubtreeRootThreshold` is high, any transaction that take up greater than 1/4 of the square they will always start at a multiple of `k` instead of having padding that might be `2k` or even `4k` (depending on what the `SubtreeRootTheshold` was).

I guess what's changed is that the multiple of `k` is encapsulated in the logic that calculates the `SubTreeWidth` (if you see what I mean)

So what I will do is take your suggestion and say that it's just a multiple of the `SubtreeWidth` but then add another section explaining how the subtree width is calculated.